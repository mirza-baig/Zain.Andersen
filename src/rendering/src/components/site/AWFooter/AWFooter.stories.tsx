// Global
import { Story, Meta } from '@storybook/react';
// Local
import AWFooter, { AWFooterProps } from './AWFooter';
import defaultData from './AWFooter.mock-data';

export default {
  title: 'Site/AWFooter',
  component: AWFooter,
} as Meta;
const Template: Story<AWFooterProps> = (props) => <AWFooter {...props} />;
export const Default = Template.bind({});
Default.args = defaultData;
