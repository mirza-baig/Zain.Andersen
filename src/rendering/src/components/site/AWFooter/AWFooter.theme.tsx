// Lib
import { ThemeFile } from 'lib/context/ThemeContext';

export const AWFooterTheme: ThemeFile = {
  aw: {
    classes: {},
  },
  rba: {
    classes: {},
  },
};
