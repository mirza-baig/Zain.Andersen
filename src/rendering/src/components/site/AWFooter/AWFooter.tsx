// Global
import { useState } from 'react';
import { Feature } from 'src/.generated/Feature.EnterpriseWeb.model';
import { withDatasourceCheck, Text } from '@sitecore-jss/sitecore-jss-nextjs';
// Components

//import { AWFooterTheme } from './AWFooter.theme';
import { LinkWrapper } from 'src/helpers/LinkWrapper';
import { SvgIcon } from 'src/helpers/SvgIcon';
import ImageWrapper from 'src/helpers/Media/ImageWrapper';
import { useCurrentScreenType } from 'lib/utils/get-screen-type';
import { getEnum } from 'lib/utils';
import { useExperienceEditor } from 'lib/utils';

export type AWFooterProps = Feature.EnterpriseWeb.AndersenWindows.Components.Navigation.AWFooter;
const AWFooter = (props: AWFooterProps) => {
  const { fields } = props;
  const isEE = useExperienceEditor();
  const { screenType } = useCurrentScreenType();
  const isDesktop = screenType !== 'sm' && screenType !== 'md' ? true : false;

  const currentYear = new Date().getFullYear().toString();
  const copyrightText = fields?.copyright?.value;
  let textWithReplacedYear = copyrightText;
  if (copyrightText?.includes('{currentYear}')) {
    textWithReplacedYear = copyrightText.replace('{currentYear}', currentYear);
  }

  const navGroup = fields?.children;
  const socialMenu =
    navGroup?.length &&
    navGroup.filter(function (nav: AWFooterProps) {
      return nav?.fields?.menuTitle?.value === 'socialMenu';
    });
  const footerMenu =
    navGroup?.length &&
    navGroup.filter(function (nav: AWFooterProps) {
      return nav.fields?.menuTitle?.value === 'footerMenu';
    });
  const privacyMenu =
    navGroup?.length &&
    navGroup.filter(function (nav: AWFooterProps) {
      return nav.fields?.menuTitle?.value === 'privacyMenu';
    });
  let privacyMenuArray = privacyMenu?.length && privacyMenu[0].fields.children;
  let socialMenuArray = socialMenu?.length && socialMenu[0].fields.children;
  const footerMenuArray = footerMenu?.length && footerMenu[0].fields.children;
  if (socialMenuArray == 0) {
    socialMenuArray = null;
  }
  if (privacyMenuArray == 0) {
    privacyMenuArray = null;
  }
  const Accordion = ({ title, children }: AWFooterProps) => {
    const [isOpen, setOpen] = useState(false);
    return (
      <div>
        <div
          className={`border-t border-solid border-t-white py-xs ${
            isOpen
              ? 'open [&_span]:after:-rotate-90 [&_span]:after:transition-all [&_span]:after:duration-[0.1s]'
              : '[&_span]:after:rotate-90 [&_span]:after:transition-all [&_span]:after:duration-[0.1s]'
          }`}
          onClick={() => setOpen(!isOpen)}
        >
          <span className="flex items-center justify-between text-small font-heavy uppercase leading-[14px] after:text-base after:content-['\276F'] ml:text-xxs ">
            {title}
          </span>
        </div>
        <div
          className={`accordion-item ${
            !isOpen
              ? 'collapsed max-h-0 transition-[max-height] duration-[0.35s] ease-[cubic-bezier(0,1,0,1)]'
              : 'h-auto max-h-[9999px] overflow-hidden transition-[max-height] duration-[0.3s] ease-[cubic-bezier(1,0,1,0)]'
          }`}
        >
          <div>{children}</div>
        </div>
      </div>
    );
  };

  if (!fields) {
    return null;
  }

  return (
    <div
      data-component="site/awfooter"
      className="awfooter bg-black py-m text-base text-white ml:py-l"
    >
      <div className="px-m ml:max-w-screen-lg lg:mx-auto">
        <div className="flex flex-col flex-wrap ml:flex-row">
          <div className="flex flex-col max-ml:order-4 max-ml:mt-l ml:w-[16.6%] ml:basis-[16.6%] ml:pr-xxs">
            <div className="mb-s font-sans text-xs font-heavy uppercase max-ml:hidden ">
              <Text tag={'h3'} field={fields.tagLine} />
            </div>
            <div className="mb-s text-body max-ml:order-2">
              <Text field={{ value: textWithReplacedYear }} />
            </div>
            <div className="flex flex-wrap max-ml:order-1 max-ml:mb-s">
              {socialMenuArray &&
                socialMenuArray.map((menu: AWFooterProps, index: number) => {
                  return (
                    menu.fields?.navItemLink.value &&
                    getEnum(menu.fields?.navItemIcon) && (
                      <div key={index} className="mr-xxs mt-xxxs">
                        {isEE ? (
                          <SvgIcon icon={getEnum(menu.fields?.navItemIcon)} />
                        ) : (
                          <LinkWrapper ctaSection="footer" field={menu.fields?.navItemLink}>
                            <SvgIcon icon={getEnum(menu.fields?.navItemIcon)} />
                          </LinkWrapper>
                        )}
                      </div>
                    )
                  );
                })}
            </div>
          </div>
          <div className="flex flex-col max-ml:order-1 ml:w-[83.4%] ml:basis-[83.4%] ml:pl-xxs">
            <div className="mb-l font-sans text-sm-s font-heavy uppercase ml:hidden">
              <Text tag={'h3'} field={fields.tagLine} />
            </div>
            <div className="flex w-full max-ml:flex-col max-ml:border-b max-ml:border-solid max-ml:border-b-white ml:shrink ml:grow ml:basis-0">
              {footerMenuArray &&
                footerMenuArray.map((menu: AWFooterProps, index: number) => {
                  return (
                    <div key={index} className="flex flex-col ml:w-[20%] ml:pr-xxs">
                      {isDesktop ? (
                        <>
                          <div className="mb-xxxs font-sans text-xxs font-heavy uppercase leading-none">
                            <Text tag={'span'} field={menu.fields?.navGroupTitle} />
                          </div>
                          <ul className={`flex flex-col`}>
                            {menu.fields?.children.map((nav: AWFooterProps, index: number) => {
                              return (
                                nav.fields?.navItemLink && (
                                  <li key={index} className="py-xxxs">
                                    <LinkWrapper
                                      ctaSection="footer"
                                      field={nav.fields?.navItemLink}
                                      className="text-body hover:underline"
                                    >
                                      {nav.fields?.navItemLink.value.target === '_blank' && (
                                        <SvgIcon icon="new-tab" className="ml-xxxs inline-flex" />
                                      )}
                                    </LinkWrapper>
                                  </li>
                                )
                              );
                            })}
                          </ul>
                        </>
                      ) : (
                        <Accordion title={menu.fields?.navGroupTitle.value as string}>
                          {menu.fields && menu.fields.children.length && (
                            <ul className={`ease flex flex-col  transition-all duration-1000`}>
                              {menu.fields?.children.map((nav: AWFooterProps, index: number) => {
                                return (
                                  nav.fields?.navItemLink && (
                                    <li key={index} className="py-xxxs ">
                                      <LinkWrapper
                                        ctaSection="footer"
                                        field={nav.fields?.navItemLink}
                                        className="text-small hover:underline"
                                      >
                                        {nav.fields?.navItemLink.value.target === '_blank' && (
                                          <SvgIcon icon="new-tab" className="ml-xxxs inline-flex" />
                                        )}
                                      </LinkWrapper>
                                    </li>
                                  )
                                );
                              })}
                            </ul>
                          )}
                        </Accordion>
                      )}
                    </div>
                  );
                })}
            </div>
          </div>

          <div className="mt-l flex flex-col max-ml:order-3 max-ml:mb-l ml:w-[16.6%] ml:basis-[16.6%] ml:pr-xxs">
            {fields?.logoCTA?.value?.href !== '' ? (
              <LinkWrapper ctaSection="footer" field={fields.logoCTA}>
                <ImageWrapper
                  image={fields.logo}
                  additionalMobileClasses="max-w-[120px]"
                  additionalDesktopClasses="max-w-[120px]"
                />
              </LinkWrapper>
            ) : (
              <ImageWrapper
                image={fields.logo}
                additionalMobileClasses="max-w-[120px]"
                additionalDesktopClasses="max-w-[120px]"
              />
            )}
          </div>
          <div className="mt-l flex flex-col max-ml:order-3 ml:w-[83.4%] ml:basis-[83.4%] ml:pl-xxs">
            <div className="mb-s flex w-full text-body ml:mb-xxs">
              <Text tag={'h3'} field={fields.privacyCaption} />
            </div>
            <div className="w-full flex-wrap max-ml:text-body md:flex">
              {privacyMenuArray &&
                privacyMenuArray.map((menu: AWFooterProps, index: number) => {
                  return (
                    menu.fields?.navItemLink && (
                      <span key={index}>
                        <LinkWrapper
                          ctaSection="footer"
                          className="text-body underline"
                          field={menu.fields?.navItemLink}
                        >
                          {menu.fields?.navItemLink.value.target === '_blank' && (
                            <SvgIcon icon="new-tab" className="ml-xxxs inline-flex" />
                          )}
                        </LinkWrapper>
                        {index < privacyMenuArray.length - 1 && (
                          <span className="inline-flex md:px-xxxs"> | </span>
                        )}
                      </span>
                    )
                  );
                })}
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

export default withDatasourceCheck()<AWFooterProps>(AWFooter);
