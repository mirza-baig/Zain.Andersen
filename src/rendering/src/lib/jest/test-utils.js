// Global
import { render, fireEvent, waitFor } from '@testing-library/react';
import { ComponentPropsContext } from '@sitecore-jss/sitecore-jss-nextjs';
import { FastSitecoreContext } from 'lib/overrides/FastSitecoreContext';

// Lib
import { createSampleComponentFactory } from 'lib/mocks/mock-placeholder';

export const eeSitecoreContext = {
  route: {
    name: 'Foo',
    placeholders: {},
  },
  context: {
    pageEditing: true,
    site: {
      name: 'JssTestWeb',
    },
    language: 'en',
  },
};

export const renderComponent = (
  Component,
  {
    componentProps = {},
    staticProps = {},
    componentFactory = createSampleComponentFactory(),
    sitecoreContext = {
      route: {
        name: 'Foo',
        placeholders: {},
      },
    },
  }
) => {
  const mockLayoutData = {
    sitecore: sitecoreContext,
  };

  return render(
    <ComponentPropsContext value={staticProps}>
      <FastSitecoreContext componentFactory={componentFactory} layoutData={mockLayoutData}>
        <Component {...componentProps} />
      </FastSitecoreContext>
    </ComponentPropsContext>
  );
};

export const snapshot = (Component, mockData) => {
  const component = renderComponent(Component, mockData);
  expect(component.container.firstChild).toMatchSnapshot();
  return component;
};

export const hasDataComponent = async (component, name) => {
  await waitFor(() => {
    const nodes = component.baseElement.querySelectorAll(`[data-component="${name}"]`);
    return expect(nodes.length).toBe(1);
  });
};

export const mockDateAs = (epochMs) => {
  if (typeof epochMs !== 'number') {
    console.error(
      'mockDateAs only accepts epoch values in milliseconds to prevent timezones from factoring in to Date creation.'
    );
  }

  jest.spyOn(global.Date, 'now').mockImplementation(() => new Date(epochMs).valueOf());
};

export const testFieldError = async (Component, mockData, { label, value, error }) => {
  const component = renderComponent(Component, mockData);
  const field = component.getByLabelText(label);
  fireEvent.change(field, { target: { value } });
  field.blur();
  await waitFor(() => {
    const err = component.queryByText(error);
    return expect(err).not.toBe(null);
  });
};
