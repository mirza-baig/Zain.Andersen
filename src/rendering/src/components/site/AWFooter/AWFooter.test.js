// Global
import { hasDataComponent, snapshot } from 'lib/jest/test-utils';
// Local
import AWFooter from './AWFooter';
import defaultData from './AWFooter.mock-data';

it('renders correctly', async () => {
  const component = snapshot(AWFooter, { componentProps: defaultData });
  await hasDataComponent(component, 'site/awfooter');
});
