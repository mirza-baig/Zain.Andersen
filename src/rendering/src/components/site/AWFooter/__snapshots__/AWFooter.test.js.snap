// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  class="awfooter bg-black py-m text-base text-white ml:py-l"
  data-component="site/awfooter"
>
  <div
    class="px-m ml:max-w-screen-lg lg:mx-auto"
  >
    <div
      class="flex flex-col flex-wrap ml:flex-row"
    >
      <div
        class="flex flex-col max-ml:order-4 max-ml:mt-l ml:w-[16.6%] ml:basis-[16.6%] ml:pr-xxs"
      >
        <div
          class="mb-s font-sans text-xs font-heavy uppercase max-ml:hidden "
        >
          <h3>
            LOVE THE LIFE YOU SEE
          </h3>
        </div>
        <div
          class="mb-s text-body max-ml:order-2"
        >
          &copy; 2023 ANDERSEN CORPORATION. ALL RIGHTS RESERVED.
        </div>
        <div
          class="flex flex-wrap max-ml:order-1 max-ml:mb-s"
        >
          <div
            class="mr-xxs mt-xxxs"
          >
            <a
              aria-label=""
              class=""
              data-gtm-click=""
              data-gtm-dl-event="nav_click"
              data-gtm-dl-nav-section="footer"
              href="/"
              role="link"
              tabindex="0"
              target=""
              title=""
            >
              
              <span />
            </a>
          </div>
          <div
            class="mr-xxs mt-xxxs"
          >
            <a
              aria-label=""
              class=""
              data-gtm-click=""
              data-gtm-dl-event="nav_click"
              data-gtm-dl-nav-section="footer"
              href="/"
              role="link"
              tabindex="0"
              target=""
              title=""
            >
              
              <span />
            </a>
          </div>
          <div
            class="mr-xxs mt-xxxs"
          >
            <a
              aria-label=""
              class=""
              data-gtm-click=""
              data-gtm-dl-event="nav_click"
              data-gtm-dl-nav-section="footer"
              href="/"
              role="link"
              tabindex="0"
              target=""
              title=""
            >
              
              <span />
            </a>
          </div>
          <div
            class="mr-xxs mt-xxxs"
          >
            <a
              aria-label=""
              class=""
              data-gtm-click=""
              data-gtm-dl-event="nav_click"
              data-gtm-dl-nav-section="footer"
              href="/"
              role="link"
              tabindex="0"
              target=""
              title=""
            >
              
              <span />
            </a>
          </div>
          <div
            class="mr-xxs mt-xxxs"
          >
            <a
              aria-label=""
              class=""
              data-gtm-click=""
              data-gtm-dl-event="nav_click"
              data-gtm-dl-nav-section="footer"
              href="/"
              role="link"
              tabindex="0"
              target=""
              title=""
            >
              
              <span />
            </a>
          </div>
          <div
            class="mr-xxs mt-xxxs"
          >
            <a
              aria-label=""
              class=""
              data-gtm-click=""
              data-gtm-dl-event="nav_click"
              data-gtm-dl-nav-section="footer"
              href="/"
              role="link"
              tabindex="0"
              target=""
              title=""
            >
              
              <span />
            </a>
          </div>
          <div
            class="mr-xxs mt-xxxs"
          >
            <a
              aria-label=""
              class=""
              data-gtm-click=""
              data-gtm-dl-event="nav_click"
              data-gtm-dl-nav-section="footer"
              href="/"
              role="link"
              tabindex="0"
              target=""
              title=""
            >
              
              <span />
            </a>
          </div>
        </div>
      </div>
      <div
        class="flex flex-col max-ml:order-1 ml:w-[83.4%] ml:basis-[83.4%] ml:pl-xxs"
      >
        <div
          class="mb-l font-sans text-sm-s font-heavy uppercase ml:hidden"
        >
          <h3>
            LOVE THE LIFE YOU SEE
          </h3>
        </div>
        <div
          class="flex w-full max-ml:flex-col max-ml:border-b max-ml:border-solid max-ml:border-b-white ml:shrink ml:grow ml:basis-0"
        >
          <div
            class="flex flex-col ml:w-[20%] ml:pr-xxs"
          >
            <div
              class="mb-xxxs font-sans text-xxs font-heavy uppercase leading-none"
            >
              <span>
                About Andersen
              </span>
            </div>
            <ul
              class="flex flex-col"
            >
              <li
                class="py-xxxs"
              >
                <a
                  aria-label="Item Link"
                  class="text-body hover:underline"
                  data-gtm-click=""
                  data-gtm-dl-event="nav_click"
                  data-gtm-dl-nav-section="footer"
                  href="/"
                  role="link"
                  tabindex="0"
                  target=""
                  title=""
                >
                  Item Link
                </a>
              </li>
              <li
                class="py-xxxs"
              >
                <a
                  aria-label="Item Link"
                  class="text-body hover:underline"
                  data-gtm-click=""
                  data-gtm-dl-event="nav_click"
                  data-gtm-dl-nav-section="footer"
                  href="/"
                  role="link"
                  tabindex="0"
                  target=""
                  title=""
                >
                  Item Link
                </a>
              </li>
            </ul>
          </div>
          <div
            class="flex flex-col ml:w-[20%] ml:pr-xxs"
          >
            <div
              class="mb-xxxs font-sans text-xxs font-heavy uppercase leading-none"
            >
              <span>
                RbA
              </span>
            </div>
            <ul
              class="flex flex-col"
            >
              <li
                class="py-xxxs"
              >
                <a
                  aria-label="Item Link"
                  class="text-body hover:underline"
                  data-gtm-click=""
                  data-gtm-dl-event="nav_click"
                  data-gtm-dl-nav-section="footer"
                  href="/"
                  role="link"
                  tabindex="0"
                  target=""
                  title=""
                >
                  Item Link
                </a>
              </li>
            </ul>
          </div>
          <div
            class="flex flex-col ml:w-[20%] ml:pr-xxs"
          >
            <div
              class="mb-xxxs font-sans text-xxs font-heavy uppercase leading-none"
            >
              <span>
                Explore Products
              </span>
            </div>
            <ul
              class="flex flex-col"
            >
              <li
                class="py-xxxs"
              >
                <a
                  aria-label="Item Link"
                  class="text-body hover:underline"
                  data-gtm-click=""
                  data-gtm-dl-event="nav_click"
                  data-gtm-dl-nav-section="footer"
                  href="/"
                  role="link"
                  tabindex="0"
                  target=""
                  title=""
                >
                  Item Link
                </a>
              </li>
            </ul>
          </div>
          <div
            class="flex flex-col ml:w-[20%] ml:pr-xxs"
          >
            <div
              class="mb-xxxs font-sans text-xxs font-heavy uppercase leading-none"
            >
              <span>
                Get Started
              </span>
            </div>
            <ul
              class="flex flex-col"
            >
              <li
                class="py-xxxs"
              >
                <a
                  aria-label="Item Link"
                  class="text-body hover:underline"
                  data-gtm-click=""
                  data-gtm-dl-event="nav_click"
                  data-gtm-dl-nav-section="footer"
                  href="/"
                  role="link"
                  tabindex="0"
                  target=""
                  title=""
                >
                  Item Link
                </a>
              </li>
            </ul>
          </div>
          <div
            class="flex flex-col ml:w-[20%] ml:pr-xxs"
          >
            <div
              class="mb-xxxs font-sans text-xxs font-heavy uppercase leading-none"
            >
              <span>
                Find Help
              </span>
            </div>
            <ul
              class="flex flex-col"
            >
              <li
                class="py-xxxs"
              >
                <a
                  aria-label="Item Link"
                  class="text-body hover:underline"
                  data-gtm-click=""
                  data-gtm-dl-event="nav_click"
                  data-gtm-dl-nav-section="footer"
                  href="/"
                  role="link"
                  tabindex="0"
                  target=""
                  title=""
                >
                  Item Link
                </a>
              </li>
            </ul>
          </div>
        </div>
      </div>
      <div
        class="mt-l flex flex-col max-ml:order-3 max-ml:mb-l ml:w-[16.6%] ml:basis-[16.6%] ml:pr-xxs"
      >
        <a
          aria-label=""
          class=""
          data-gtm-click=""
          data-gtm-dl-event="nav_click"
          data-gtm-dl-nav-section="footer"
          href="/"
          role="link"
          tabindex="0"
          target=""
          title=""
        >
          
          <div
            class="relative hidden md:block aspect-picture    max-w-[120px]"
          >
            <span
              style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 14.122533748701974% 0px 0px 0px;"
              />
              <img
                alt="Andersen Windows Logo"
                data-nimg="responsive"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
          </div>
          <div
            class="relative w-full md:hidden max-w-[120px]"
          >
            <span
              style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 14.122533748701974% 0px 0px 0px;"
              />
              <img
                alt="Andersen Windows Logo"
                data-nimg="responsive"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover; object-position: center;"
              />
              <noscript />
            </span>
          </div>
        </a>
      </div>
      <div
        class="mt-l flex flex-col max-ml:order-3 ml:w-[83.4%] ml:basis-[83.4%] ml:pl-xxs"
      >
        <div
          class="mb-s flex w-full text-body ml:mb-xxs"
        >
          <h3>
            Andersen collects certain categories of personal information. See links for more information.
          </h3>
        </div>
        <div
          class="w-full flex-wrap max-ml:text-body md:flex"
        >
          <span>
            <a
              aria-label="Terms"
              class="text-body underline"
              data-gtm-click=""
              data-gtm-dl-event="nav_click"
              data-gtm-dl-nav-section="footer"
              href="/"
              role="link"
              tabindex="0"
              target=""
              title=""
            >
              Terms
            </a>
            <span
              class="inline-flex md:px-xxxs"
            >
               | 
            </span>
          </span>
          <span>
            <a
              aria-label="Privacy Policy "
              class="text-body underline"
              data-gtm-click=""
              data-gtm-dl-event="nav_click"
              data-gtm-dl-nav-section="footer"
              href="/"
              role="link"
              tabindex="0"
              target=""
              title=""
            >
              Privacy Policy 
            </a>
            <span
              class="inline-flex md:px-xxxs"
            >
               | 
            </span>
          </span>
          <span>
            <a
              aria-label="Privacy Notice for CA Residents "
              class="text-body underline"
              data-gtm-click=""
              data-gtm-dl-event="nav_click"
              data-gtm-dl-nav-section="footer"
              href="/"
              role="link"
              tabindex="0"
              target=""
              title=""
            >
              Privacy Notice for CA Residents 
            </a>
            <span
              class="inline-flex md:px-xxxs"
            >
               | 
            </span>
          </span>
          <span>
            <a
              aria-label=" EEO Policy "
              class="text-body underline"
              data-gtm-click=""
              data-gtm-dl-event="nav_click"
              data-gtm-dl-nav-section="footer"
              href="/"
              role="link"
              tabindex="0"
              target=""
              title=""
            >
               EEO Policy 
            </a>
            <span
              class="inline-flex md:px-xxxs"
            >
               | 
            </span>
          </span>
          <span>
            <a
              aria-label=" Opt Out of Sale or Sharing of Personal Information"
              class="text-body underline"
              data-gtm-click=""
              data-gtm-dl-event="nav_click"
              data-gtm-dl-nav-section="footer"
              href="/"
              role="link"
              tabindex="0"
              target=""
              title=""
            >
               Opt Out of Sale or Sharing of Personal Information
            </a>
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
